Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -1788,9 +1832,7 @@ fileprivate extension UIViewController {
 // MARK: - SettingsViewModel delegation
 extension StatusTableViewController: SettingsViewModelDelegate {
     func dosingEnabledChanged(_ value: Bool) {
-        DispatchQueue.main.async {
-            self.deviceManager.loopManager.settings.dosingEnabled = value
-        }
```

Comment:
[nit] consider adding
```
        dispatchPrecondition(condition: .onQueue(.main))
```

Reply:
All UI code has this assumption in it. I think dispatchPrecondition should be reserved for those situations where violation is a special concern; do you think this code falls into that category?

Reply:
If setting `self.deviceManager.loopManager.settings.dosingEnabled` requires being on `main` then, yes, please add the `dispatchPrecondition(condition: .onQueue(.main))`.  If not, don't bother.

Reply:
In code that is considered "UI code", like code on a view controller, or general Loop level code, 99% of the time, the executing queue will be the main queue, and all the variable reads/writes will be required to be on the main queue. If we put dispatch conditions in for all of those, it would be a mess. For just this pattern alone (updating a settings property) there are 36 existing cases, none of which have a dispatchPrecondition on. I think we should reserve this defensive coding strategy for those areas that have special protections.